homeassistant/components/zabbix/__init__.py

```diff
@@ -159,7 +163,25 @@ def event_to_metrics(
             )
             metrics.append(metric)
 
-        string_keys.update(strings)
+        if include_strings:
+            string_keys_count = len(string_keys)
+            string_keys.update(strings)
+            if len(string_keys) != string_keys_count:
```

Comment:
```suggestion
            if len(strings) > 0:
                string_keys.update(strings)
```
this also works, right?

Reply:
Yes and no - discovery only needs to be updated if new keys are detected - the length of the keys is a heuristic for that. If we change it the way you suggest the discovery is always updated - which is not required.

Reply:
Can you elaborate a but on this please, I don't understand the problem.